### PR TITLE
apache-drill: update livecheck

### PR DESCRIPTION
**Pull request by chenrui333**
Currently, the `brew livecheck` checks against Apache ftp server, but it is not an accurate tracking for the release.

This PR is to update the tracking to check against the homepage. 

---

relates to #60732

### CODE REVIEW

**Comment by nandahkrishna** [comment]
I read @issyl0's comment in the PR you referenced, but I'm just concerned they haven't added the release note _yet_ – 1.18 released on 4 September 2020 so it may well be that they just haven't gotten around to creating the release note. Does that seem valid?

File: Formula/apache-drill.rb [hunk 1]
```diff
@@ -7,7 +7,8 @@ class ApacheDrill < Formula
   license "Apache-2.0"
 
   livecheck do
-    url :stable
+    url "https://drill.apache.org/download/"
+    regex(/Drill v?(\d+(?:\.\d+)+) was released/i)
```
Comment (samford) [inline Formula/apache-drill.rb:11]:
```suggestion
    /href=.*?apache-drill[._-]v?(\d+(?:\.\d+)+)\.t/i
```

**Review by samford — DISMISSED** [review]
It _is_ strange that version 1.18 is available for download and has also been released on GitHub but hasn't been announced. However, I know this has happened with other formulae before as well, so it's not unheard of. There are times where the release artifacts are published but the release isn't official until there's an announcement on the first-party website (or the download links are referenced there). In those cases, we make sure to check the first-party website, so that livecheck doesn't report a new version until it's official.

It's hard to say for sure but this may be another one of those cases. I'm in favor of checking the first-party download page here and we can always modify this again if it doesn't turn out to be the correct decision.

That said, I would modify the regex to match against the tarball links on the downloads page instead, as that may be more reliable than the "Drill 1.17 was released" text.

**Comment by chenrui333** [comment]
Yeah, I am surprised as well, but just to be cautious with this.

This actually happened with `clojure` recently too.

**Comment by SMillerDev** [comment]
We _could_ just ask upstream if it's officially released and if it's going to be on the website.

**Comment by chenrui333** [comment]
go for the merge? looks like the 1.18 release still did not get announced yet.

**Comment by SMillerDev** [comment]
Did we ask upstream?

**Comment by chenrui333** [comment]
not quite yet, let me do it now.

**Comment by nandahkrishna** [comment]
Luckily for us, someone seems to have asked upstream about the same thing [here](http://mail-archives.apache.org/mod_mbox/drill-user/202009.mbox/%3cCANxXE_+5+2dDBSnECr2LGFD-rh63aVGCiyKk+E4=641gttD_iw@mail.gmail.com%3e). Apparently everything except the release notes are done.

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
surprisingly, apache-drill still did not get a chance to write the release blog.

**Pull request merged**